app/questionnaire/questionnaire_schema.py

```diff
@@ -713,7 +713,7 @@ def get_summary_title_for_section(self, section_id: str) -> str | None:
 
     def show_summary_on_completion_for_section(self, section_id: str) -> bool | None:
         if summary := self.get_summary_for_section(section_id):
-            return summary.get("show_on_completion", False)
+            return bool(summary.get("show_on_completion"))
```

Comment:
I think based off type hint guidance this ought to be the original code with a type ignore